### PR TITLE
Bug 1873218: Do not fail verify connection if there is no local uaid

### PR DESCRIPTION
**Pull request by tarikeshaq**
### Pull Request checklist ###

- **Breaking changes**:  This PR follows our [breaking change policy](https://github.com/mozilla/application-services/blob/main/docs/howtos/breaking-changes.md)
  - [x] This PR follows the breaking change policy:
     - This PR has no breaking API changes, or
     - There are corresponding PRs for our consumer applications that resolve the breaking changes and have been approved
- [x] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [x] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

File: components/push/src/internal/push_manager.rs [hunk 1]
```diff
@@ -226,7 +226,10 @@ impl<Co: Connection, Cr: Cryptography, S: Storage> PushManager<Co, Cr, S> {
         if force_verify {
             self.verify_connection_rate_limiter.reset(&self.store);
         }
-        if !self.verify_connection_rate_limiter.check(&self.store) {
+
+        // If we were rate limited or there are no subscriptions yet, we should signal to the
+        // consumer that everything is ok
+        if !self.verify_connection_rate_limiter.check(&self.store) || self.uaid.is_none() {
```
Comment (bendk) [inline components/push/src/internal/push_manager.rs:232]: Should this check count towards the rate limiter if we don't actually make a request?  I'm mostly wondering about this and don't have a strong opinion, what do you think?

Comment (tarikeshaq) [reply components/push/src/internal/push_manager.rs:232]: Nope, good catch - the order of the || is off!

Comment (tarikeshaq) [reply components/push/src/internal/push_manager.rs]: and on that note, I'm openning a ticket to re-evaluate our rate-limiting strategy here, the interface has side-effects, and it's not clear the persistence to disk is exactly correct when it comes to local clocks

**Review by bendk — APPROVED** [review]
Looks great!

File: components/push/src/internal/push_manager.rs (hunk 1 above)
Comment (tarikeshaq) [reply components/push/src/internal/push_manager.rs]: https://bugzilla.mozilla.org/show_bug.cgi?id=1873237

**Pull request merged**